### PR TITLE
Add mutation to clear transform caches

### PR DESCRIPTION
**Pull request by wolfs**
We also enable the chrome trace for Gradle 5.x

### CODE REVIEW

File: src/main/java/org/gradle/profiler/mutations/ClearArtifactTransformCacheMutator.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		public Configurator(File gradleUserHome) {
+			super(gradleUserHome);
+		}
+
+		@Override
+		protected BuildMutator newInstance(File gradleUserHome, AbstractCleanupMutator.CleanupSchedule schedule) {
+			return new ClearArtifactTransformCacheMutator(gradleUserHome, schedule);
+		}
+	}
+}
```
Comment (lptr) [inline src/main/java/org/gradle/profiler/mutations/ClearArtifactTransformCacheMutator.java]: Let's reformat to use spaces. :/ Can you please add an `.editorconfig` to ensure we don't add more code with tabs?

File: subprojects/chrome-trace/src/main/java/org/gradle/trace/listener/BuildOperationListenerAdapter.java [hunk 2]
```diff
@@ -18,7 +18,7 @@ static BuildOperationListenerAdapter create(Gradle gradle, TraceResult traceResu
         if (gradleVersion.matches("4\\.[0-6](\\D.*)?")) {
             return new Gradle40BuildOperationListenerAdapter(gradle, new Gradle40BuildOperationListenerInvocationHandler(traceResult));
         }
-        if (gradleVersion.startsWith("4.")) {
+        if (gradleVersion.startsWith("4.") || gradleVersion.startsWith("5.")) {
```
Comment (lptr) [inline subprojects/chrome-trace/src/main/java/org/gradle/trace/listener/BuildOperationListenerAdapter.java]: This sounds flimsy. Shouldn't this be `> 4.0`? I.e. add a `VersionNumber` type that can parse and compare version numbers and use that?

File: subprojects/chrome-trace/src/main/java/org/gradle/trace/listener/BuildOperationListenerAdapter.java [hunk 3]
```diff
@@ -1,28 +1,35 @@
...
-        if (gradleVersion.equals("4.0-milestone-2")) {
+        if (gradleVersion.equals(version("4.0-milestone-2"))) {
             return new Gradle35BuildOperationListenerAdapter(gradle, new Gradle40BuildOperationListenerInvocationHandler(traceResult));
         }
-        if (gradleVersion.matches("4\\.[0-6](\\D.*)?")) {
+        if (inVersionRange(gradleVersion, "4.0", "4.8")) {
             return new Gradle40BuildOperationListenerAdapter(gradle, new Gradle40BuildOperationListenerInvocationHandler(traceResult));
         }
-        if (gradleVersion.startsWith("4.") || gradleVersion.startsWith("5.")) {
+        if (gradleVersion.compareTo(version("4.8")) >= 0) {
```
Comment (lptr) [inline subprojects/chrome-trace/src/main/java/org/gradle/trace/listener/BuildOperationListenerAdapter.java]: What happened to 4.7? Shouldn't `Gradle47BuildOperationListenerInvocationHandler` be used on that version, too?

Comment (wolfs) [reply subprojects/chrome-trace/src/main/java/org/gradle/trace/listener/BuildOperationListenerAdapter.java]: Yes, I failed to parse the regex correctly... Fixed.

File: subprojects/chrome-trace/src/main/java/org/gradle/trace/listener/BuildOperationListenerAdapter.java [hunk 4]
```diff
@@ -1,28 +1,35 @@
...
             return new Gradle47BuildOperationListenerAdapter(gradle, new Gradle47BuildOperationListenerInvocationHandler(traceResult));
         }
         throw new IllegalStateException("Gradle version " + gradleVersion + " not supported, 3.3+ required");
     }
 
+    static boolean inVersionRange(GradleVersion gradleVersion, String lower, String upper) {
+        return gradleVersion.compareTo(version(lower)) >= 0 && gradleVersion.compareTo(version(upper)) < 0;
+    }
```
[the remark this answers is not in the document]
Comment (wolfs) [reply subprojects/chrome-trace/src/main/java/org/gradle/trace/listener/BuildOperationListenerAdapter.java]: If we make the upper limit inclusive, we need to know which Gradle versions had patch releases. That's why I made the bound exclusive.

I wrote out the comparisons, but the verbose Java `compare` calls made it worse IMO.

**Comment by lptr** [comment]
LGTM

**Comment by wolfs** [comment]
@lptr I added an integration test. PTAL!

**Pull request merged**